### PR TITLE
chore: Add objectParser for ObjectTypeAnnotation in Parse Server options

### LINKED ISSUE #9907: PARSE_SERVER_AUTH_PROVIDERS is passed as string instead of object — built-in adapters (e.g. Microsoft) throw “options are required”
### New Issue Checklist

- [x]  I am reporting a non-security issue.
- [x]  I agree that any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE)
- [x]  I have searched [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue) and found no duplicates.

### Issue Description

When configuring authentication providers via the environment variable PARSE_SERVER_AUTH_PROVIDERS, Parse Server receives the value as a string instead of a parsed object.
As a result, built-in adapters such as the Microsoft one throw:

Error: Microsoft options are required.

The root cause is that the CLI definitions for auth lack an objectParser action, so the JSON string is not parsed when building the configuration from environment variables.

### Steps to reproduce

Run Parse Server using only environment variables (no config file).
Example with Docker:

```
docker run --rm -p 1337:1337 \
  -e PARSE_SERVER_APPLICATION_ID=app \
  -e PARSE_SERVER_MASTER_KEY=master \
  -e PARSE_SERVER_DATABASE_URI=mongodb://host.docker.internal:27017/parse \
  -e PARSE_SERVER_URL=http://localhost:1337/parse \
  -e PARSE_SERVER_AUTH_PROVIDERS='{"microsoft":{"clientId":"id","clientSecret":"secret"}}' \
  parseplatform/parse-server:8.4.0
```

Add a simple Cloud Function:

```
Parse.Cloud.define('debugAuthProviders', async (req) => {
  return { typeOfAuth: typeof req.config.auth };
});
```

Call it via REST API:

```
curl -X POST http://localhost:1337/parse/functions/debugAuthProviders \
  -H "X-Parse-Application-Id: app" \
  -H "Content-Type: application/json"
```

### Actual Outcome

Response:

`{ "result": { "typeOfAuth": "string" } }`

Then, when using authData.microsoft, Parse Server throws:

`Error: Microsoft options are required.`

### Expected Outcome

req.config.auth should be an object, not a string.
Auth adapters (Microsoft, Google, etc.) should receive the correct provider options from the parsed environment variable.

### Environment

Server
- Parse Server version: 8.4.0
- Node.js version: 14.19.2
- Operating system: Linux (Azure App Service / Docker)
- Deployment: Azure Web App (containerized)

Database
- System: MongoDB
- Version: 8.0
- Host: Azure Cosmos DB (Mongo API)

Client

- SDK: JavaScript
- SDK version: 7.0.2

### Logs
```
error: Uncaught internal server error. Microsoft options are required. 
{"stack":"Error: Microsoft options are required.\n
    at MicrosoftAdapter.validateOptions (/parse-server/lib/Adapters/Auth/BaseCodeAuthAdapter.js:18:13)\n
    at loadAuthAdapter (/parse-server/lib/Adapters/Auth/index.js:162:13)\n
    at Object.getValidatorForProvider (/parse-server/lib/Adapters/Auth/index.js:182:25)\n
    at /parse-server/lib/Auth.js:439:44"}
```

### Additional Context & Proposed Fix

The environment variable PARSE_SERVER_AUTH_PROVIDERS is correctly set and valid JSON.

However, auth is declared in src/Options/index.js as a plain object type, so the generator does not include an objectParser in Definitions.js.

### Suggested change
```
 export interface AuthAdapter {
   enabled: ?boolean;
 }
+
+export interface AuthProvidersOptions {
+  [string]: AuthAdapter;
+}

-  auth: ?{ [string]: AuthAdapter };
+  auth: ?AuthProvidersOptions;

```

This causes the generated Definitions.js to include:

```
auth: {
  env: 'PARSE_SERVER_AUTH_PROVIDERS',
  action: parsers.objectParser,
  default: {}
}
```

✅ Backward compatible
✅ Fixes adapter initialization
✅ Matches the documented “stringified JSON” expectation

I’m happy to open a PR with the change and related tests once confirmed.

### PR DESCRIPTION
**Pull request by mattia1208**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue
Closes: https://github.com/parse-community/parse-server/issues/9907

The issue occurs because the `mapperFor` function in `buildConfigDefinitions.js` did not include a specific case for the `ObjectTypeAnnotation` type in Flow definitions.  
As a result, configuration options defined as Flow objects - such as:
`auth: ?{ [string]: AuthAdapter };`
were not mapped to the appropriate parser (objectParser) when generating CLI definitions.

This caused environment variables like PARSE_SERVER_AUTH_PROVIDERS to be treated as raw strings instead of being parsed as JSON objects.
In practice, this broke the built-in Microsoft authentication adapter (and any other adapter relying on object-type auth configs), throwing errors such as:
`Error: Microsoft options are required.`
The missing objectParser action in the generated definitions.js file prevented the environment variable from being correctly parsed into an object structure.

## Approach

* Updated the `mapperFor` function in `buildConfigDefinitions.js` to handle `ObjectTypeAnnotation` by mapping it to the `objectParser`, ensuring object types are parsed correctly.
* Added the `objectParser` action to the `auth` option in `ParseServerOptions` in `Definitions.js`, enabling proper parsing of the `auth` configuration as an object.

## Tasks

- [x]  Add fix for ObjectTypeAnnotation handling in mapperFor
- [x]  Regenerate CLI definitions
- [x]  Add objectParser action for auth option
- [x] Add tests

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved parsing of authentication provider configuration so JSON object inputs from environment variables are recognized and deserialized correctly.
  * Enhanced type-to-parser mapping to correctly handle object-shaped configuration entries.

* **Tests**
  * Added unit tests validating type-to-parser mappings for various configuration shapes and edge cases.

* **Chores**
  * Prevents build/transformation script from running when modules are imported.
  * Exposed a helper for testing to simplify unit test coverage.

### CODE REVIEW

**Comment by parseplatformorg** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/acinader/pr-checks/462e3b27-9708-4217-96c2-8fe95aaf29fa?refs=8ea3f4cf0db859a8b5cc02bb96c94fb9e4b3b1c3&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

**Comment by mattia1208** [comment]
@coderabbitai review

**Comment by mattia1208** [comment]
@coderabbitai review

**Comment by mattia1208** [comment]
Hi @mtrezza,
from my side this is ready.
Please let me know if you need any changes or adjustments before merging.

Thanks!

File: resources/buildConfigDefinitions.js [hunk 1]
```diff
@@ -372,12 +374,18 @@ This code has been generated by resources/buildConfigDefinitions.js
...
-const babel = require('@babel/core');
-const res = babel.transformFileSync('./src/Options/index.js', {
-  plugins: [plugin, '@babel/transform-flow-strip-types'],
-  babelrc: false,
-  auxiliaryCommentBefore,
-  sourceMaps: false,
-});
-require('fs').writeFileSync('./src/Options/Definitions.js', res.code + '\n');
-require('fs').writeFileSync('./src/Options/docs.js', docs);
+// Only run the transformation when executed directly, not when imported by tests
```
Comment (mtrezza) [inline resources/buildConfigDefinitions.js:377]: What's the reason for this change?

Comment (mtrezza) [reply resources/buildConfigDefinitions.js:377]: @mattia1208 friendly ping

Comment (mattia1208) [reply resources/buildConfigDefinitions.js:377]: @mtrezza for me it wasn’t necessary, it was requested by coderabbitai.

Comment (mtrezza) [reply resources/buildConfigDefinitions.js:377]: @coderabbitai This looks strange. Does the code still work after this change?

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [8.6.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/8.6.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [8.6.0](https://github.com/parse-community/parse-server/releases/tag/8.6.0)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [8.6.0](https://github.com/parse-community/parse-server/releases/tag/8.6.0)
